dataland-backend/src/main/kotlin/org/dataland/datalandbackend/interfaces/DataStoreInterface.kt

```diff
@@ -1,62 +1,20 @@
 package org.dataland.datalandbackend.interfaces
 
-import org.dataland.datalandbackend.model.CompanyMetaInformation
-import org.dataland.datalandbackend.model.DataIdentifier
-import org.dataland.datalandbackend.model.DataSetMetaInformation
-import org.dataland.datalandbackend.model.StorableDataSet
-
 /**
  * Defines the required functionalities for a data store
  */
 interface DataStoreInterface {
     /**
-     * Method to add a data set to the data store
-     * @param storableDataSet contains
-     * companyId ID of the company the data belongs to
-     * dataType as the type of data (e.g. EU Taxonomy)
-     * data as data to be stored in the data store
+     * Method to insert data into to the data store
+     * @param data contains the data to be stored in the data store
      * @return ID of the newly created data store entry
      */
-    fun addDataSet(storableDataSet: StorableDataSet): String
-
-    /**
-     * Method to list the meta information of all data sets in the data store
-     * @return list of meta information of all data sets in the data store
-     */
-    fun listDataSets(): List<DataSetMetaInformation>
+    fun insertDataSet(data: String): String
 
     /**
      * Method to get the data of a single entry in the data store
-     * @param dataIdentifier identifier of the stored data (consists of data id and data type)
-     * @return all data associated to the identifier provided
-     */
-    fun getStorableDataSet(dataIdentifier: DataIdentifier): StorableDataSet
-
-    /**
-     * Method to add a company to the data store
-     * @param companyName name of the company to be stored in the data store
-     * @return meta information of the newly created entry in the data store (companyId and companyName)
-     */
-    fun addCompany(companyName: String): CompanyMetaInformation
-
-    /**
-     * Method to list matching companies in the data store by searching for a name
-     * @param name string used for substring matching against the companyName of all entries
-     * @return list of all matching companies in the data store
-     */
-    fun listCompaniesByName(name: String): List<CompanyMetaInformation>
-
-    /**
-     * Method to list all existing data sets of a given company
-     * @param companyId
-     * @return list of all data set IDs along with their data type
-     */
-    fun listDataSetsByCompanyId(companyId: String): List<DataIdentifier>
-
-    /**
-     * Method to retrieve meta information about a specific company
-     * @param companyId
-     * @return meta information consisting of company Id and company name
+     * @param dataId identifier of the stored data
+     * @return all data associated to the dataId provided
      */
-    fun getCompanyById(companyId: String): CompanyMetaInformation
+    fun selectDataSet(dataId: String): String
```

Comment:
den Namen finde ich unglücklich. Aus dem store kommt ja erstmal nur ein String zurück mit den nur den Daten. Wie wäre es mit getData und oben entsprechen insertData

Reply:
Ich finde in unserem Projekt/Use Case manchmal verwirrend, wenn Methoden-Namen im Code doppelt vorkommen und dabei auch noch voneinander abhängig sind bzw. gemeinsam in Code Blöcken verwendet werden.
Z.B. hat der DataManager bereits eine Methode `getDataSet`. 

Das war der Grund, warum ich die "datenbank-ähnlichen" Bezeichnungen `selectDataSet` und `insertDataSet` für den InMemoryDataStore gewählt habe, um sie explizit von den DataManager-Methoden `getDataSet` und `addDataSet` abzuheben.

Daher mein Gegenvorschlag:  Im InMemoryDataStore `selectData` und `insertData`, und im DataManager `getData` und `addData` ?

Oder vllt im DataManager `getDataFromStore` und `addDataToStore` ?

Lass uns darüber sprechen